packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -58,38 +81,65 @@ class FlutterGoldenFileComparator implements GoldenFileComparator {
   /// The [goldens] and [defaultComparator] parameters are visible for testing
   /// purposes only.
   static Future<FlutterGoldenFileComparator> fromDefaultComparator({
-    GoldensClient goldens,
+    dynamic goldens,
     LocalFileComparator defaultComparator,
   }) async {
     defaultComparator ??= goldenFileComparator;
 
-    // Prepare the goldens repo.
-    goldens ??= GoldensClient();
-    await goldens.prepare();
+    const Platform platform = LocalPlatform();
+    if (!_testingWithSkiaGold(platform)) {
+      // Prepare the goldens repo.
+      goldens ??= GoldensClient();
+      await goldens.prepare();
+    } else {
+      goldens ??= SkiaGoldClient();
+    }
 
     // Calculate the appropriate basedir for the current test context.
-    final FileSystem fs = goldens.fs;
+    const FileSystem fs = LocalFileSystem();
     final Directory testDirectory = fs.directory(defaultComparator.basedir);
     final String testDirectoryRelativePath = fs.path.relative(testDirectory.path, from: goldens.flutterRoot.path);
-    return FlutterGoldenFileComparator(goldens.repositoryRoot.childDirectory(testDirectoryRelativePath).uri);
+    return FlutterGoldenFileComparator(goldens.comparisonRoot.childDirectory(testDirectoryRelativePath).uri);
   }
 
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
+    SkiaGoldClient _skiaClient;
+    if (_testingWithSkiaGold(_platform)) {
+      _skiaClient = SkiaGoldClient();
+      golden = _addPrefix(golden);
+      await update(golden, imageBytes);
+    }
+
     final File goldenFile = _getGoldenFile(golden);
     if (!goldenFile.existsSync()) {
       throw TestFailure('Could not be compared against non-existent file: "$golden"');
     }
-    final List<int> goldenBytes = await goldenFile.readAsBytes();
-    // TODO(tvolkert): Improve the intelligence of this comparison.
-    if (goldenBytes.length != imageBytes.length) {
-      return false;
+
+    if (_testingWithSkiaGold(_platform)) {
+      final bool authorized = await _skiaClient.auth(fs.directory(basedir));
```

Comment:
Do we need to re-authenticate for every single compare? Can we do this just once?